src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs

```diff
@@ -36,15 +34,19 @@ public KestrelServer(IFeatureCollection features, KestrelServerOptions options,
                 throw new ArgumentNullException(nameof(applicationLifetime));
             }
 
-            if (logger == null)
+            if (loggerFactory == null)
             {
-                throw new ArgumentNullException(nameof(logger));
+                throw new ArgumentNullException(nameof(loggerFactory));
             }
 
+            Options = options.Value ?? new KestrelServerOptions();
             _applicationLifetime = applicationLifetime;
-            _logger = logger;
-            Features = features;
-            Options = options;
+            _logger = loggerFactory.CreateLogger("Microsoft.AspNetCore.Server.Kestrel");
+            Features = new FeatureCollection();
+            var componentFactory = new HttpComponentFactory(Options);
+            Features.Set<IHttpComponentFactory>(componentFactory);
```

Comment:
It seems to be replaceable and will then be used for stream and header pooling. I agree this is a strange place to put it. Why not just resolve it from DI?

Regardless, this is unrelated to the current changes. I just moved it from ServerFactory.